hydromt/models/model.py

```diff
@@ -570,7 +487,6 @@ def write_config(
         self, config_name: Optional[str] = None, config_root: Optional[str] = None
     ):
         """Write config to <root/config_fn>."""
-        self._assert_write_mode()
```

Comment:
Same here replace instead of deleting so that what this is moved to components this part won't be forgotten but replaced in another way if appropriate.
Same comments for all other places where this was removed

Reply:
I've added them back